accessible/toolboxview.component.js

```diff
@@ -0,0 +1,136 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a toolbox is rendered
+ * in AccessibleBlockly. Also handles any interactions with the toolbox.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.ToolboxView = ng.core
+.Component({
+  selector: 'toolbox-view',
+  template: `
+    <h3 #toolboxTitle id="blockly-toolbox-title">Toolbox</h3>
+    <ol #tree id="blockly-toolbox-tree" *ngIf="makeArray(sightedToolbox) && makeArray(sightedToolbox).length > 0" tabIndex="0" role="group" class="blocklyTree" [attr.aria-labelledby]="toolboxTitle.id" (keydown)="treeService.keyHandler($event, tree)">
+    {{treeService.setActiveAttribute(tree)}}
+    <li #parent *ngIf="toolboxHasCategories" role="treeitem" aria-level="1" aria-selected=false *ngFor="#category of makeArray(sightedToolbox); #i=index">
+      <label #name>{{category.attributes.name.value}}</label>
+      {{labelCategory(name, i, tree)}}
+      <ol  role="group" *ngIf="getToolboxWorkspace(category).topBlocks_.length > 0">
+        <toolbox-tree-view *ngFor="#block of getToolboxWorkspace(category).topBlocks_" [level]=2 [block]="block" [displayBlockMenu]="true" [clipboardService]="sharedClipboardService"></toolbox-tree-view>
+        {{addClass(parent, "blocklyHasChildren")}}
+      </ol>
+    </li>
+    <div *ngIf="!toolboxHasCategories">
+      <toolbox-tree-view *ngFor="#block of getToolboxWorkspace(toolboxCategories[0]).topBlocks_; #i=index" [level]=1 [block]="block" [displayBlockMenu]="true" [clipboardService]="sharedClipboardService" [index]="i" [tree]="tree"></toolbox-tree-view>
+    </div>
+    </ol>
+  `,
+  directives: [blocklyApp.ToolboxTreeView],
+})
+.Class({
+  constructor: [blocklyApp.TreeService, function(_treeService) {
+    var _this = this;
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function() {
+      if (xhttp.readyState == 4 && xhttp.status == 200) {
+        _this.sightedToolbox = xhttp.responseXML;
+      }
+    };
+    xhttp.open('GET', blocklyApp.gameManager
+      .levelToolboxes[blocklyApp.gameManager.level], true);
+    xhttp.send();
+
+    this.toolboxCategories = [];
+    this.toolboxWorkspaces = {};
+    this.treeService = _treeService;
+
+    this.toolboxHasCategories = false;
+  }],
+  labelCategory: function(h2, i, tree) {
+    var parent = h2.parentNode;
+    while (parent != null && parent.tagName != 'LI') {
+      parent = parent.parentNode;
+    }
+    parent.setAttribute('aria-label', h2.innerText);
+    parent.id = 'toolbox-tree-node' + i;
+    if (i == 0 && tree.getAttribute('aria-activedescendant') ==
+      'toolbox-tree-node0') {
+      this.addClass(parent, 'blocklyActiveDescendant');
+      this.treeService.setActiveDesc(parent, tree.id);
+      parent.setAttribute('aria-selected', 'true');
+    }
+  },
+  addClass: function(node, classText) {
```

Comment:
I believe the addClass conversation is happening in an early file. If we do end up needing it, I'll be sure to put it in utilsService or something.